xmtp_mls/src/groups/members.rs

```diff
@@ -24,39 +23,32 @@ impl MlsGroup {
 
     pub fn members_with_provider(
         &self,
-        provider: impl OpenMlsProvider,
+        provider: &XmtpOpenMlsProvider,
     ) -> Result<Vec<GroupMember>, GroupError> {
         let openmls_group = self.load_mls_group(provider)?;
-        aggregate_member_list(&openmls_group)
-    }
-}
+        // TODO: Replace with try_into from extensions
+        let group_membership = extract_group_membership(openmls_group.extensions())?;
+        let requests = group_membership
+            .members
+            .into_iter()
+            .map(|(inbox_id, sequence_id)| (inbox_id, sequence_id as i64))
+            .collect();
 
-pub fn aggregate_member_list(openmls_group: &OpenMlsGroup) -> Result<Vec<GroupMember>, GroupError> {
-    let member_map: HashMap<String, GroupMember> = openmls_group
-        .members()
-        .filter_map(|member| {
-            let basic_credential = BasicCredential::try_from(&member.credential).ok()?;
-            Identity::get_validated_account_address(
-                basic_credential.identity(),
-                &member.signature_key,
-            )
-            .ok()
-            .map(|account_address| (account_address, member.signature_key.clone()))
-        })
-        .fold(
-            HashMap::new(),
-            |mut acc, (account_address, signature_key)| {
-                acc.entry(account_address.clone())
-                    .and_modify(|e| e.installation_ids.push(signature_key.clone()))
-                    .or_insert(GroupMember {
-                        account_address,
-                        installation_ids: vec![signature_key],
-                    });
-                acc
-            },
-        );
+        let conn = provider.conn_ref();
+        let association_state_map = StoredAssociationState::batch_read_from_cache(conn, &requests)?;
+        // TODO: Figure out what to do with missing members from the local DB. Do we go to the network? Load from identity updates?
```

Comment:
Should we make this into an additional ticket?

Reply:
https://github.com/xmtp/libxmtp/issues/750